lib/src/simulations/merge_sort.dart

```diff
@@ -0,0 +1,318 @@
+      if (sort) await _mergeSort(leftIndex, middleIndex);
+      if (sort) await _mergeSort(middleIndex + 1, rightIndex);
+      if (sort) await addContainer(barwidth, Theme.of(context).primaryColor, 0, _numberOfElements - 1);
+      if (sort) await merge(leftIndex, middleIndex, rightIndex);
+    }
+  }
+
+  makeGreen() async {
+    if (pause)
+      for (var i = 0; i < _numberOfElements; i++) {
+        await Future.delayed(Duration(milliseconds: delay2));
+
+        containers[i] = Container(
+          color: Colors.greenAccent[400],
+          height: (elements[i] + 0.5),
+          width: barwidth,
+        );
+        streamContainer.add(containers);
+      }
+  }
+
+  _sort() async {
+    setState(() {
+      sort = true;
+    });
+    await _mergeSort(0, elements.length - 1);
+    setState(() {
+      sort = false;
+    });
+    if (!pause) makeGreen();
+  }
+
+  _randomize() {
+    elements = [];
+    containers = [];
+    for (int i = 0; i < _numberOfElements; i++) {
+      elements.add(Random().nextInt(400));
+    }
+    for (int i = 0; i < _numberOfElements; i++) {
+      containers.add(Container(
+        color: Theme.of(context).primaryColor,
+        height: (elements[i] + 0.5),
+        width: MediaQuery.of(context).size.width / (elements.length + 1),
+      ));
+    }
+
+    sort = false;
+    streamContainer.add(containers);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      // ignore: missing_return
+      builder: (_, BoxConstraints constraints) {
+        if (constraints.maxWidth != 0) {
+          ScreenUtil.init(
+            constraints,
+            designSize: Size(512.0, 1024.0),
+            allowFontScaling: true,
+          );
+          return Scaffold(
+            appBar: AppBar(
+              automaticallyImplyLeading: false,
+              leading: IconButton(
+                icon: Icon(Icons.arrow_back_ios),
+                onPressed: () {
+                  sort = false;
+                  Navigator.pop(context);
+                },
+              ),
+              centerTitle: true,
+              title: Text(
+                'Merge Sort',
+                style: Theme.of(context).textTheme.headline6,
+              ),
+            ),
+            floatingActionButton: FloatingActionButton(
+              backgroundColor: Colors.white,
+              child: (!sort)
+                  ? Icon(
+                      Icons.play_arrow,
+                      color: Colors.black,
+                    )
+                  : Icon(
+                      Icons.pause,
+                      color: Colors.black,
+                    ),
+              onPressed: () {
+                pause = !pause;
+                //TODO: Fix the bug, i.e, when we pause the code shows unusual behaviour
+                if (!sort)
+                  _sort();
+                else
+                  setState(() {
+                    sort = false;
+                  });
+              },
```

Comment:
The pause bug needs to be fixed, when paused, it restarts the sort.

Reply:
Ok

Reply:
Yes, it actually scans the array from the starting and I tried making some other changes but I wasn't able to implement the pause function properly.